crates/holochain_integrity_types/src/info.rs

```diff
@@ -187,3 +223,33 @@ pub trait EnumVariantLen<const V: u8> {
     /// The ending point of this variant.
     const ENUM_VARIANT_LEN: u8 = Self::ENUM_VARIANT_START + Self::ENUM_VARIANT_INNER_LEN;
 }
+
+/// Helper trait for types that are internally
+/// represented as [`u8`] but need to be used
+/// as indicies into containers.
+pub trait U8Index {
+    /// Get the [`usize`] index from this type.
+    fn index(&self) -> usize;
+}
+
+impl U8Index for ZomeDependencyIndex {
+    fn index(&self) -> usize {
+        self.0 as usize
+    }
+}
+impl U8Index for EntryDefIndex {
```

Comment:
What do you think about renaming `EntryDefIndex` to just `EntryType`? It seems like it better illustrates that it's similar to `LinkType`, and it makes more sense to be part of `ZomeTypes`. Then the "entry type" is an index that gives you the "entry def".

Reply:
We already have a type named `EntryType` though

Reply:
I'd like to change this to `EntryType`. Maybe we can change the name of the enum we currently call `EntryType`. Either way it feels like a follow up PR?